core/src/main/java/org/infinispan/configuration/cache/ClusteringConfiguration.java

```diff
@@ -100,4 +100,28 @@ public L1Configuration l1() {
    public StateTransferConfiguration stateTransfer() {
       return stateTransferConfiguration;
    }
+
+   @Override
+   public boolean matches(ClusteringConfiguration other) {
+      if (this.cacheMode == CacheMode.LOCAL) {
+         return other.cacheMode == CacheMode.LOCAL;
+      }
+
+      if (!attributes.matches(other.attributes)) return false;
+      if (children.length != other.children.length) return false;
+      for (int i = 0; i < children.length; i++) {
+         ConfigurationElement ours = children[i];
+         ConfigurationElement theirs = other.children[i];
+
+         // Hash configuration does not take num-owners into account for replicated caches.
```

Comment:
I feel like this should exist in the HashConfiguration class instead of here. Then we don't have to do this whole ignored method as well.

Reply:
Yeah, I can create a specific match method in HashConfiguration. I did add it here because the HashConfiguration itself doesn't know whether it is replicated. We need to verify if replicated here and invoke a specific match method.